exchanges/deribit/deribit_wrapper.go

```diff
@@ -0,0 +1,1276 @@
+func (d *Deribit) GetFeeByType(ctx context.Context, feeBuilder *exchange.FeeBuilder) (float64, error) {
+	if feeBuilder == nil {
+		return 0, fmt.Errorf("%T %w", feeBuilder, common.ErrNilPointer)
+	}
+	if !d.AreCredentialsValid(ctx) && // Todo check connection status
+		feeBuilder.FeeType == exchange.CryptocurrencyTradeFee {
+		feeBuilder.FeeType = exchange.OfflineTradeFee
+	}
+	var fee float64
+	var err error
+	switch feeBuilder.FeeType {
+	case exchange.CryptocurrencyTradeFee:
+		fee, err = calculateTradingFee(feeBuilder)
+		if err != nil {
+			return 0, err
+		}
+	case exchange.CryptocurrencyDepositFee:
+	case exchange.CryptocurrencyWithdrawalFee:
+		// Withdrawals are processed instantly if the balance in our hot wallet permits so. We keep only a small percentage of coins in hot storage,
+		// therefore there is a chance that your withdrawal cannot be processed immediately. If needed, once a day we will replenish the balance of the hot wallet from the cold storage.
+	case exchange.OfflineTradeFee:
+		fee = getOfflineTradeFee(feeBuilder.PurchasePrice, feeBuilder.Amount)
+	}
+	if fee < 0 {
+		fee = 0
+	}
+	return fee, nil
+}
+
+// ValidateAPICredentials validates current credentials used for wrapper
+// functionality
+func (d *Deribit) ValidateAPICredentials(ctx context.Context, assetType asset.Item) error {
+	_, err := d.UpdateAccountInfo(ctx, assetType)
+	return d.CheckTransientError(err)
+}
+
+// GetHistoricCandles returns candles between a time period for a set time interval
+func (d *Deribit) GetHistoricCandles(ctx context.Context, pair currency.Pair, a asset.Item, interval kline.Interval, start, end time.Time) (*kline.Item, error) {
+	req, err := d.GetKlineRequest(pair, a, interval, start, end, false)
+	if err != nil {
+		return nil, err
+	}
+	intervalString, err := d.GetResolutionFromInterval(interval)
+	if err != nil {
+		return nil, err
+	}
+	switch a {
+	case asset.Futures, asset.Spot:
+		var tradingViewData *TVChartData
+		if d.Websocket.IsConnected() {
+			tradingViewData, err = d.WSRetrievesTradingViewChartData(d.formatFuturesTradablePair(req.RequestFormatted), intervalString, start, end)
+		} else {
+			tradingViewData, err = d.GetTradingViewChart(ctx, d.formatFuturesTradablePair(req.RequestFormatted), intervalString, start, end)
+		}
+		if err != nil {
+			return nil, err
+		} else if len(tradingViewData.Ticks) == 0 {
+			return nil, kline.ErrNoTimeSeriesDataToConvert
+		}
+		checkLen := len(tradingViewData.Ticks)
+		if len(tradingViewData.Open) != checkLen ||
+			len(tradingViewData.High) != checkLen ||
+			len(tradingViewData.Low) != checkLen ||
+			len(tradingViewData.Close) != checkLen ||
+			len(tradingViewData.Volume) != checkLen {
+			return nil, fmt.Errorf("%s - %v: invalid trading view chart data received", a, req.RequestFormatted)
+		}
+		listCandles := make([]kline.Candle, len(tradingViewData.Ticks))
+		for x := range tradingViewData.Ticks {
+			listCandles[x] = kline.Candle{
```

Comment:
You are not filtering results by start/end time and so this is getting errors in `exchange_wrapper_standards` testing. You may also need to set it to UTC

This may not resolve all the `exchange_wrapper_standards` tests, so please check those out after next commit